cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -398,5 +456,14 @@ private void publishAudit(Id.DatasetInstance datasetInstance, AuditType auditTyp
     // TODO: Add properties to Audit Payload (CDAP-5220)
     AuditPublishers.publishAudit(auditPublisher, datasetInstance, auditType, AuditPayload.EMPTY_PAYLOAD);
   }
+
+  private void ensureAccess(DatasetId datasetId) throws Exception {
```

Comment:
you should still document that this throws UnauthorizedException. That is what callers can expect when the user is not authorized, and that is important to know.